File: ver_v4/start_peer_node.go
package main

import (
	"log"
	"os"
	"os/exec"
)

func main() {
	commandA := "./pr_make_tree" // 매개변수는 참여노드의 IP와 port (2개)
	commandB := "./pr_node"

	cmdA := exec.Command(commandA)
	cmdB := exec.Command(commandB)

	cmdA.Stdout = os.Stdout
	cmdA.Stderr = os.Stderr

	cmdB.Stdout = os.Stdout
	cmdB.Stderr = os.Stderr

	if err := cmdA.Start(); err != nil {
		log.Fatal(err)
	}

	if err := cmdB.Start(); err != nil {
		log.Fatal(err)
	}

	if err := cmdA.Wait(); err != nil {
		log.Fatal(err)
	}

	if err := cmdB.Wait(); err != nil {
		log.Fatal(err)
	}
}

File: ver_v4/start_peer_node.py
import os
import argparse
import subprocess
import time

current_dir = os.path.dirname(os.path.abspath(__file__))

command_a = os.path.join(current_dir, "pr_make_tree")  # "./pr_make_tree" 대신 경로를 지정
command_b = os.path.join(current_dir, "pr_node")  # "./pr_node" 대신 경로를 지정

process_a = subprocess.Popen(command_a, shell=True)
process_b = subprocess.Popen(command_b, shell=True)

# 각 프로세스의 종료를 기다림
process_a.wait()
process_b.wait()
